pyuvdata/uvfits.py

```diff
@@ -158,15 +158,8 @@ def read_uvfits(self, filename, run_check=True, run_check_acceptability=True):
 
         # get shapes
         self.Nfreqs = hdr.pop('NAXIS4')
-        if self.data_array.shape[2] != self.Nfreqs:
```

Comment:
Because it's impossible to get them. It turns out that you can't write a fits file that would be broken in these ways.